apps/examples/ble_perfs/ble_perfs_main.c

```diff
@@ -0,0 +1,452 @@
+}
+
+static void utc_cb_desc_b_1(ble_server_attr_cb_type_e type, ble_conn_handle conn_handle, ble_attr_handle attr_handle, void *arg)
+{
+	char *arg_str = "None";
+	if (arg != NULL) {
+		arg_str = (char *)arg;
+	}
+	BLE_LOGD( "[DESC_A_1][%s] type : %d / handle : %d / attr : %02x \n", arg_str, type, conn_handle, attr_handle);
+}
+
+
+static void perfs_ble_reset_data(void)
+{
+	g_perf_data.packet_size = 0;
+	g_perf_data.packet_count = 0;
+
+	g_perf_data.start_time.tv_sec = 0;
+	g_perf_data.start_time.tv_usec = 0;
+	g_perf_data.end_time.tv_sec = 0;
+	g_perf_data.end_time.tv_usec = 0;
+}
+
+static int perfs_ble_get_result(void)
+{
+	struct timeval diff_time = {0, 0};
+	double total_time = 0;
+	double speed = 0;
+
+	BLE_LOGD( "======== Bandwidth Analysis =================\n");
+        BLE_LOGD( "start time: %ld sec %ld usec \n", g_perf_data.start_time.tv_sec, g_perf_data.start_time.tv_usec);
+        BLE_LOGD( "end time: %ld sec %ld usec \n", g_perf_data.end_time.tv_sec, g_perf_data.end_time.tv_usec);
+
+	diff_time.tv_sec = g_perf_data.end_time.tv_sec - g_perf_data.start_time.tv_sec;
+	diff_time.tv_usec = g_perf_data.end_time.tv_usec - g_perf_data.start_time.tv_usec;
+	if(diff_time.tv_usec < 0) {
+		diff_time.tv_sec = diff_time.tv_sec - 1;
+		diff_time.tv_usec = diff_time.tv_usec + 1000000;
+	}
+
+        BLE_LOGD( "diff time: %ld sec %ld usec \n", diff_time.tv_sec, diff_time.tv_usec);
+	
+	total_time = diff_time.tv_sec + (double) diff_time.tv_usec / 1000000;
+	speed = g_perf_data.packet_size * g_perf_data.packet_count * 8 / total_time;
+
+	BLE_LOGD( "=================================================\n");
+	BLE_LOGD( "Packet Count  : % d \n", g_perf_data.packet_count);
+	BLE_LOGD( "Packet Size   : % d byte\n", g_perf_data.packet_size);
+	BLE_LOGD( "Total time    : %.2lf ms \n", total_time*1000);
+	BLE_LOGD( "data received : %ld byte \n", g_perf_data.packet_size * g_perf_data.packet_count);
+	BLE_LOGD( "Bandwidth     : %.2lf Kbps \n", speed / 1024);
+	BLE_LOGD( "=================================================\n");
+	
+	return 0;
+}
+
+static void ble_peri_cb_charact_rmc_sync(ble_server_attr_cb_type_e type, ble_conn_handle conn_handle, ble_attr_handle attr_handle, void* arg) 
+{
+	uint8_t buf[256] = { 0, };
+	ble_data blue_data = { buf, sizeof(buf) };
+	struct perfs_data_t *p_data = (struct perfs_data_t *)arg;
+
+	ble_result_e ret = ble_server_attr_get_data(attr_handle, &blue_data);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		BLE_LOGD( "[RMC_SYNC] Fail to get attr data\n");
+		return;
+	}
+
+	if (blue_data.data[0] == FIRST_PACKET) {
+		perfs_ble_reset_data();	
+		p_data->packet_size = blue_data.length;
+        	gettimeofday(&p_data->start_time, NULL);
+        	BLE_LOGD( "start time: %ld sec %ld usec \n", p_data->start_time.tv_sec, p_data->start_time.tv_usec);
+	}
+
+	p_data->packet_count ++;
+
+	if (type == BLE_SERVER_ATTR_CB_WRITING_NO_RSP || type == BLE_SERVER_ATTR_CB_WRITING) {
+		if(blue_data.data[0] == blue_data.data[p_data->packet_size - 1]) {
```

Comment:
Please leave a charcter space after "is", "for", "while" etc

Reply:
done